instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -32,69 +36,69 @@
       new ExecutionAttribute<>(TracingExecutionInterceptor.class.getName() + ".Scope");
   static final ExecutionAttribute<AwsSdkRequest> AWS_SDK_REQUEST_ATTRIBUTE =
       new ExecutionAttribute<>(TracingExecutionInterceptor.class.getName() + ".AwsSdkRequest");
+  static final ExecutionAttribute<SdkHttpRequest> SDK_HTTP_REQUEST_ATTRIBUTE =
+      new ExecutionAttribute<>(TracingExecutionInterceptor.class.getName() + ".SdkHttpRequest");
 
   static final String COMPONENT_NAME = "java-aws-sdk";
 
-  private final AwsSdkHttpClientTracer tracer;
+  private final Instrumenter<ExecutionAttributes, SdkHttpResponse> instrumenter;
   private final boolean captureExperimentalSpanAttributes;
   private final FieldMapper fieldMapper;
 
   TracingExecutionInterceptor(
-      AwsSdkHttpClientTracer tracer, boolean captureExperimentalSpanAttributes) {
-    this.tracer = tracer;
+      Instrumenter<ExecutionAttributes, SdkHttpResponse> instrumenter,
+      boolean captureExperimentalSpanAttributes) {
+    this.instrumenter = instrumenter;
     this.captureExperimentalSpanAttributes = captureExperimentalSpanAttributes;
-    fieldMapper = new FieldMapper();
-  }
-
-  @Override
-  public void beforeExecution(
-      Context.BeforeExecution context, ExecutionAttributes executionAttributes) {
-    io.opentelemetry.context.Context parentOtelContext = io.opentelemetry.context.Context.current();
-    if (!tracer.shouldStartSpan(parentOtelContext)) {
-      return;
-    }
-    io.opentelemetry.context.Context otelContext =
-        tracer.startSpan(parentOtelContext, executionAttributes);
-    executionAttributes.putAttribute(CONTEXT_ATTRIBUTE, otelContext);
-    if (executionAttributes
-        .getAttribute(SdkExecutionAttribute.CLIENT_TYPE)
-        .equals(ClientType.SYNC)) {
-      // We can only activate context for synchronous clients, which allows downstream
-      // instrumentation like Apache to know about the SDK span.
-      executionAttributes.putAttribute(SCOPE_ATTRIBUTE, otelContext.makeCurrent());
-    }
+    this.fieldMapper = new FieldMapper();
   }
 
   @Override
   public SdkHttpRequest modifyHttpRequest(
       Context.ModifyHttpRequest context, ExecutionAttributes executionAttributes) {
+    SdkHttpRequest httpRequest = context.httpRequest();
+
     io.opentelemetry.context.Context otelContext = getContext(executionAttributes);
     if (otelContext == null) {
-      return context.httpRequest();
+      return httpRequest;
     }
 
-    SdkHttpRequest.Builder builder = context.httpRequest().toBuilder();
-    tracer.inject(otelContext, builder);
+    SdkHttpRequest.Builder builder = httpRequest.toBuilder();
+    AwsXrayPropagator.getInstance().inject(otelContext, builder, AwsSdkInjectAdapter.INSTANCE);
     return builder.build();
   }
 
   @Override
   public void afterMarshalling(
```

Comment:
Oops, I thought `afterMarshaling` runs after `modifyHttpRequest`! Can you move this method above `modifyHttpRequest` for less chance of misunderstanding?